### PR TITLE
editoast: add `utoipa` annotation for `infra/{id}/switch_types`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of #4580

### CODE REVIEW

File: editoast/src/views/infra/mod.rs [hunk 1]
```diff
@@ -374,15 +375,23 @@ async fn rename(
 }
 
 /// Return the railjson list of switch types
+#[utoipa::path(
+    tag = "infra",
+    params(InfraIdParam),
+    responses(
+        (status = 200, description = "A list of switch types", body = Vec<SwitchType>,  example = json!({"id": "Point", "ports": ["LEFT", "RIGHT", "BASE"], "groups": { "LEFT": { "src": "BASE", "dst": "LEFT" }, "RIGHT": { "src": "BASE", "dst": "RIGHT" }}})),
```
Comment (leovalais) [inline editoast/src/views/infra/mod.rs]: Better move the example to the `SwitchType` schema directly.

**Review by leovalais — APPROVED** [review]
One small change and 🚀 , thanks!

File: editoast/src/views/infra/mod.rs (hunk 1 above)
Comment (hamz2a) [reply editoast/src/views/infra/mod.rs]: Done.

File: editoast/editoast_schemas/src/infra/switch_type.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ editoast_common::schemas! {
 #[derive(Debug, Derivative, Clone, Deserialize, Serialize, PartialEq, Eq, ToSchema)]
 #[serde(deny_unknown_fields)]
 #[derivative(Default)]
+#[schema(example = json!({"id": "Point", "ports": ["LEFT", "RIGHT", "BASE"], "groups": { "LEFT": { "src": "BASE", "dst": "LEFT" }, "RIGHT": { "src": "BASE", "dst": "RIGHT" }}}))]
```
Comment (woshilapin) [inline editoast/editoast_schemas/src/infra/switch_type.rs]: minor: Do you think it's possible to help a little bit the formatting here? I believe `rustfmt` doesn't format code inside macro's call (`json!()` here). But maybe we can do it a little bit manually?

**Review by woshilapin — APPROVED** [review]

File: editoast/editoast_schemas/src/infra/switch_type.rs (hunk 2 above)
Comment (hamz2a) [reply editoast/editoast_schemas/src/infra/switch_type.rs]: Done.

File: front/src/common/api/osrdEditoastApi.ts [hunk 3]
```diff
@@ -1984,6 +1985,17 @@ export type PathfindingInput = {
...
 };
+export type SwitchPortConnection = {
+  dst: string;
+  src: string;
+};
+export type SwitchType = {
+  groups: {
+    [key: string]: SwitchPortConnection[];
+  };
+  id: string;
```
Comment (SharglutDev) [inline front/src/common/api/osrdEditoastApi.ts:1996]: In the editor, we have a type for the switch possible ids : 
```
type TrackNodeTypeId =
  | 'crossing'
  | 'single_slip_switch'
  | 'double_slip_switch'
  | 'link'
  | 'point_switch';
```
Are you checking somewhere in the back that the front sends you one of these ids ? If not, should you have an enum for that ?

Comment (SharglutDev) [reply front/src/common/api/osrdEditoastApi.ts:1996]: As discussed, we will keep that for another PR (maybe)

**Review by SharglutDev — APPROVED** [review]
Lgtm

**Pull request merged**